File: server/models/Specialty.js
const mongoose = require('mongoose');

const specialtySchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Tên chuyên khoa là bắt buộc'],
    unique: true,
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  icon: {
    type: String,
    trim: true
  },
  imageUrl: {
    type: String,
    trim: true
  },
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true
});

const Specialty = mongoose.model('Specialty', specialtySchema);

module.exports = Specialty; 

File: hospital_mobile_app/lib/domain/entities/specialty.dart
import 'package:equatable/equatable.dart';

class Specialty extends Equatable {
  final String id;
  final String name;
  final String? description;
  final String? icon;
  final String? imageUrl;
  final int doctorCount;

  const Specialty({
    required this.id,
    required this.name,
    this.description,
    this.icon,
    this.imageUrl,
    required this.doctorCount,
  });

  @override
  List<Object?> get props => [id, name, description, icon, imageUrl, doctorCount];
}
